### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by tnptop**
My suggestions on code style, placement, and some improvements in logic

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -1,6 +1,8 @@
+'use strict';
+
 // Dependencies
 const { token } = require('./config.json');
```
Comment (tnptop) [inline main.js:4]: Usually `.json` is omitted, but quite a number of people prefer to leave it there for clarity.

File: main.js [hunk 2]
```diff
@@ -1,6 +1,8 @@
+'use strict';
```
Comment (tnptop) [inline main.js:1]: Starts **every** `.js` file with `'use strict'`
Read more [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Strict_mode).

File: main.js [hunk 3]
```diff
@@ -1,6 +1,8 @@
...
+
 // Dependencies
 const { token } = require('./config.json');
-const bot = require('./src/bot.js');
```
Comment (tnptop) [inline main.js:3]: No need to specify `.js` in require expression.
Weirdly enough this is the only place I found, so it's likely that you forgot to remove it.

File: src/bot.js [hunk 4]
```diff
@@ -12,21 +14,22 @@ const bot = new Discord.Client({
 });
 
 // Import commands
-const commands = require('./commands/index');
```
Comment (tnptop) [inline src/bot.js:15]: `index.js` is required implicitly if the require path points to a folder.

These are equivalent:
```javascript
require('./commands')
require('./commands/index')
require('./commands/index.js')
```

File: src/bot.js [hunk 5]
```diff
@@ -12,21 +14,22 @@ const bot = new Discord.Client({
...
 
 bot.once('ready', async () => {
   const guild = bot.guilds.cache.get(config.guildId);
+
   // Remove all commands (To flush out unused commands)
   const collectedCommands = await guild.commands.fetch();
-  collectedCommands.forEach(async (command) => {
+  for (const command of collectedCommands) {
     await guild.commands.delete(command);
-  });
```
Comment (tnptop) [inline src/bot.js:24]: A better way to handle asynchronous looping is using `for...of` syntax (details [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of)).

File: src/commands/game/songcheck.js [hunk 6]
```diff
@@ -1,59 +1,61 @@
+'use strict';
+
 // Dependencies
 const config = require('../../../config');
 const Discord = require('discord.js');
 
-module.exports = {
```
Comment (tnptop) [inline src/commands/game/songcheck.js:5]: Avoid putting everything inside `module.exports` as it reads harder.

`module.exports` should only define what is exported; no need to include the content in there.

This pretty much applies to all command files, as seen below.

File: src/commands/game/songcheck.js [hunk 7]
```diff
@@ -1,59 +1,61 @@
...
-  config: {
-    name: 'songcheck',
-    description: 'Send a message and ask participants to voice ownership of the given song',
-    options: [{
```
Comment (tnptop) [inline src/commands/game/songcheck.js:9]: Formatting multi-line objects inside an array can be done usually in these two ways:
```javascript
// option 1
let arr = [{
  key: 'value',
  key2: 'value2',
}, {
  key: 'value',
  key2: 'value2',
}]

// option 2
let arr = [
  {
    key: 'value',
    key2: 'value2',
  },
  {
    key: 'value',
    key2: 'value2',
  },
]
```
For now I chose option 1, but you're more than welcome to change to option 2

File: src/helpers.js [hunk 8]
```diff
@@ -0,0 +1,37 @@
+'use strict';
```
Comment (tnptop) [inline src/helpers.js:1]: As `parsePack` relates directly to our use cases, my suggestion is to put it in `helpers.js` file.

File: main.js (hunk 3 above)
Comment (joksulainen) [reply main.js:3]: Yeah i forgot to remove it

File: src/bot.js (hunk 4 above)
Comment (joksulainen) [reply src/bot.js:15]: Nice to know fact, will apply to future (if any) projects.

File: src/commands/game/songcheck.js (hunk 6 above)
Comment (joksulainen) [reply src/commands/game/songcheck.js:5]: Definitely improves clarity

File: src/commands/game/songcheck.js (hunk 7 above)
Comment (joksulainen) [reply src/commands/game/songcheck.js:9]: Most likely will stick with option one as it keeps the bracket start lines consitent based on how I position the first bracket.

**Review by joksulainen — APPROVED** [review]

**Pull request merged**